test/unit/stencil_test.rb

```diff
@@ -67,10 +67,52 @@ def test_2d_stencil
         end
 
         stencil_result_gpu = base_array_gpu.pstencil([[-1, -1], [0, -1], [0, 1], [0, 0]], 10000) do |values|
-            values[0] + values[1] + values[2] + values[3]
+            values[-1][-1] + values[0][-1] + values[0][1] + values[0][0]
         end
 
         # Compare results
         assert_equal([10000, 10000, 10000, 10000, 10000, 10000, 33, 37, 41, 10000, 10000, 73, 77, 81, 10000, 10000, 113, 117, 121, 10000, 10000, 153, 157, 161, 10000, 10000, 193, 197, 201, 10000, 10000, 233, 237, 241, 10000], stencil_result_gpu.to_a)
     end
+
+    def test_non_constant_stencil
```

Comment:
Does this also work with 2D / 3D arrays? If so, we should add a unit case. If not, we should add a comment here.